File: src/lib/types/ui/library/album/AlbumDetailView.ts
export interface AlbumDetailView {
  id: string;
  library_id: number;
  title: string;
  title_normalized: string;
  album_type: string; // album | single | ep | compilation
  musicbrainz_id: string | null;
  artist_id: string | null;
  artist: string | null;
  year: number | null;
  genre: string | null;
  notes: string | null;
  cover_url: string | null;
  thumbnail_path: string | null;
  total_tracks: number;
  total_duration: number;
  created_at: string; // ISO string depuis Rust DateTime<Utc>
  updated_at: string;
}

File: src-tauri/src/mapper/library/album/album_detail_view.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct AlbumDetailView {
    pub id: String,
    pub library_id: i64,
    pub title: String,
    pub title_normalized: String,
    pub album_type: String, // album | single | ep | compilation
    pub musicbrainz_id: Option<String>,
    pub artist_id: String, // utile pour navigation
    pub artist: String,    // nom affiché
    pub year: Option<i32>,
    pub genre: Option<String>,
    pub notes: Option<String>,
    pub cover_url: Option<String>,       // cover officielle album
    pub thumbnail_path: Option<String>,  // fallback cache track si besoin
    pub total_tracks: i64,
    pub total_duration: f64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}